consensus-engine/src/overlay/branch_overlay.rs

```diff
@@ -0,0 +1,294 @@
+        self.number_of_committees > 1 && self.membership_committees[&1].contains(&id)
+    }
+
+    fn parent_committee(&self, id: NodeId) -> Option<Committee> {
+        self.committees_by_member[&id]
+            .checked_sub(1)
+            .map(|parent_id| self.membership_committees[&parent_id].clone())
+    }
+
+    fn child_committees(&self, id: NodeId) -> Vec<Committee> {
+        if self.is_member_of_leaf_committee(id) {
+            vec![]
+        } else {
+            let parent = self.committees_by_member[&id] + 1;
+            vec![self.membership_committees[&parent].clone()]
+        }
+    }
+
+    fn leaf_committees(&self, _id: NodeId) -> Vec<Committee> {
+        vec![self.membership_committees[&(self.number_of_committees - 1)].clone()]
+    }
+
+    fn node_committee(&self, id: NodeId) -> Committee {
+        let committee_id = self.committees_by_member[&id];
+        self.membership_committees[&committee_id].clone()
+    }
+
+    fn next_leader(&self) -> NodeId {
+        self.leader.next_leader(&self.nodes)
+    }
+
+    fn super_majority_threshold(&self, id: NodeId) -> usize {
+        if self.is_member_of_leaf_committee(id) {
+            return 0;
+        }
+        let committee_size = self.root_committee().len();
+        (committee_size * 2 / 3) + 1
+    }
+
+    fn leader_super_majority_threshold(&self, id: NodeId) -> usize {
+        self.super_majority_threshold(id)
+    }
+
+    fn update_leader_selection<F, E>(&self, f: F) -> Result<Self, E>
+    where
+        F: FnOnce(Self::LeaderSelection) -> Result<Self::LeaderSelection, E>,
+    {
+        match f(self.leader.clone()) {
+            Ok(leader_selection) => Ok(Self {
+                leader: leader_selection,
+                ..self.clone()
+            }),
+            Err(e) => Err(e),
+        }
+    }
+
+    fn update_committees<F, E>(&self, f: F) -> Result<Self, E>
+    where
+        F: FnOnce(Self::CommitteeMembership) -> Result<Self::CommitteeMembership, E>,
+    {
+        f(self.committee_membership.clone()).map(|committee_membership| {
+            let settings = BranchOverlaySettings {
+                nodes: self.nodes.clone(),
+                current_leader: self.current_leader,
+                number_of_levels: self.number_of_committees,
+                leader: self.leader.clone(),
+                committee_membership,
+            };
+            Self::new(settings)
+        })
+    }
+}
+
+fn build_committee_from_nodes_with_size(
+    nodes: &[NodeId],
+    number_of_committees: usize,
+) -> (Vec<CommitteeId>, HashMap<usize, Committee>) {
+    let committee_size = nodes.len() / number_of_committees;
+    let remainder = nodes.len() % number_of_committees;
+
+    let mut committees: Vec<Committee> = (0..number_of_committees)
+        .map(|n| {
+            nodes[n * committee_size..(n + 1) * committee_size]
+                .iter()
+                .cloned()
+                .collect()
+        })
+        .collect();
```

Comment:
nitpick: I think you can do this with [`chunks_exact`](https://doc.rust-lang.org/std/primitive.slice.html#method.chunks_exact)